client/components/SingleProduct.js

```diff
@@ -1,28 +1,31 @@
-import React from 'react'; 
+import React from 'react';
 import { connect } from 'react-redux';
 import { fetchProduct } from '../store/product';
+import AllReviews from './AllReviews'
 
 class SingleProduct extends React.Component {
-    constructor(props){
+    constructor(props) {
         super(props)
         this.state = {
             product: props.product
         }
     }
 
-    componentDidMount(){
-        const id = this.props.match.params.id
-        this.props.fetchProduct(id);
+    componentDidMount() {
+        const productId = Number(this.props.match.params.id);
```

Comment:
small nit,  we could just write `+this.props.match.params.id` instead of using `Number`